### PR TITLE
Move the util functions to classes

### PR DESCRIPTION
**Pull request by tweksteen**
This pull request moves the util commands to specific classes, inheriting from BaseCommand. It includes:
- Define a new class BaseCommand
- For each utils:
  - convert the function to a class, using the run() method for its main content
  - Use the attributes instead of global variables:
    `s/logger()/self.logger`,
    `s/argv/self.argv` and
    `s/chipsec_utils._cs/self.cs`
  - add a requires_driver() method that determines if the OS driver needs to be loaded depending on the arguments and potentially other factors (platform).
  - Clean up some of the import if relevant
- Modify chipsec_utils.py to use the new classes

The diff for each command is rather verbose because of the different indentation level. Using `git diff -w` to ignore the space difference may help to review this request.

The purpose of this request is to clean up how the utils are implemented in order to ease the transition when using Chipsec without drivers.

### CODE REVIEW

File: source/tool/chipsec/utilcmd/acpi_cmd.py [hunk 1]
```diff
@@ -54,44 +50,50 @@ def acpi(argv):
...
-      
-    if ( 'list' == op ):
-        logger().log( "[CHIPSEC] Enumerating ACPI tables.." )
-        _acpi.print_ACPI_table_list()      
-    elif ( 'table' == op ):
-        if len(argv) < 4:
-            print acpi.__doc__
+
+    def requires_driver(self):
+        if len(self.argv) < 3:
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/acpi_cmd.py:56]: Please add brief comment on what this if statement is checking for. 

Something like "# driver not needed for doc command"

File: source/tool/chipsec/utilcmd/acpi_cmd.py [hunk 2]
```diff
@@ -54,44 +50,50 @@ def acpi(argv):
...
+            return False
+        return True
+
+    def run(self):
+        if len(self.argv) < 3:
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/acpi_cmd.py:61]: same as above

File: source/tool/chipsec/utilcmd/cmos_cmd.py [hunk 3]
```diff
@@ -47,46 +40,52 @@ def cmos(argv):
...
-
 
-chipsec_util.commands['cmos'] = {'func' : cmos,    'start_driver' : True, 'help' : cmos.__doc__  }
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        if 3 > len(self.argv):
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/cmos_cmd.py]: While you're at it, can you please change this reverse logic to what you have above "len(self.argv) < 3".

File: source/tool/chipsec/utilcmd/mem_cmd.py [hunk 4]
```diff
@@ -112,105 +108,111 @@ def mem(argv):
...
+
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        size = 0x100
+
+        if 3 > len(self.argv):
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/mem_cmd.py]: "len(self.argv) < 3"

File: source/tool/chipsec/utilcmd/mmio_cmd.py [hunk 5]
```diff
@@ -55,44 +54,51 @@ def mmio(argv):
...
+
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        t = time.time()
+
+        if 3 > len(self.argv):
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/mmio_cmd.py]: len(self.argv) < 3

File: source/tool/chipsec/utilcmd/msr_cmd.py [hunk 6]
```diff
@@ -50,34 +41,40 @@ def msr(argv):
...
-    if 3 > len(argv):
-        print msr.__doc__
-        return
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        if 3 > len(self.argv):
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/msr_cmd.py]: len(self.argv) < 3

File: source/tool/chipsec/utilcmd/pci_cmd.py [hunk 7]
```diff
@@ -54,64 +47,71 @@ def pci(argv):
...
-        else:
-            if 'byte' == argv[6]:
+
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        if 3 > len(self.argv):
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/pci_cmd.py]: len(self.argv) < 3

File: source/tool/chipsec/utilcmd/smbus_cmd.py [hunk 8]
```diff
@@ -44,48 +38,54 @@ def smbus(argv):
...
-            buf = _smbus.read_range( dev_addr, start_off, size )
-            logger().log( "[CHIPSEC] SMBus read: device 0x%X offset 0x%X size 0x%X" % (dev_addr, start_off, size) )
-            print_buffer( buf )
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        if 3 > len(self.argv):
```
Comment (timevortex) [inline source/tool/chipsec/utilcmd/smbus_cmd.py]: len(self.argv) < 3

**Comment by timevortex** [comment]
I've added notes in some places to replace the reverse logic of "3 > len(self.argv)" with "len(self.argv) < 3". I decided not to go through each instance but I think you should change it in at least all the run functions where you've added a requires_driver function to the class and where you're using "len(self.argv) < 3" in those functions. This way there's consistency in at least the local area.

**Comment by tweksteen** [comment]
Done. PTAL.

**Comment by timevortex** [comment]
Thanks. LGTM. 

I wish the diffs were more readable - it would make me more confident about the LGTM :)

**Pull request merged**